hooks/utils.py

```diff
@@ -326,76 +343,12 @@ def write_gui_config(
-                'The password is the admin-secret from the Juju environment. '
-                'This can be found by looking in ~/.juju/environments/{}.jenv '
-                'and searching for the password field. Note that using '
-                'juju-quickstart (https://launchpad.net/juju-quickstart) can '
-                'automate logging in, as well as other parts of installing '
-                'and starting Juju.'.format(env_name))
-        else:
-            # The Juju environment name is included in the hooks context
-            # starting from juju-core v1.18.
-            login_help = (
-                'The password for newer Juju clients can be found by locating '
-                'the Juju environment file placed in ~/.juju/environments/ '
-                'with the same name as the current environment. For example, '
-                'if you have an environment named "production", then the file '
-                'is named ~/.juju/environments/production.jenv. Look for the '
-                '"password" field in the file, or if that is empty, for the '
-                '"admin-secret". Remove the quotes from the value, and use '
-                'this to log in. The password for older Juju clients (< 1.16) '
-                'is in ~/.juju/environments.yaml, and listed as the '
-                'admin-secret for the environment you are using. Note that '
-                'using juju-quickstart '
-                '(https://launchpad.net/juju-quickstart) can automate logging '
-                'in, as well as other parts of installing and starting Juju.')
-    if not juju_core_version:
-        log('Retrieving Juju version.')
-        juju_core_version = run('jujud', '--version').strip()
 
-    context = {
-        'cached_fonts': json.dumps(cached_fonts),
-        'raw_protocol': protocol,
-        'address': unit_get('public-address'),
-        'console_enabled': json.dumps(console_enabled),
-        'login_help': json.dumps(login_help),
-        'password': json.dumps(password),
-        'api_backend': json.dumps(api_backend),
-        'readonly': json.dumps(readonly),
-        'user': json.dumps(user),
-        'protocol': json.dumps(protocol),
-        'sandbox': json.dumps(sandbox),
-        'charmworld_url': json.dumps(charmworld_url),
-        'charmstore_url': json.dumps(charmstore_url),
-        'ga_key': json.dumps(ga_key),
-        'hide_login_button': json.dumps(hide_login_button),
-        'juju_core_version': json.dumps(juju_core_version),
-        'juju_env_uuid': json.dumps(juju_env_uuid),
-    }
-    if config_js_path is None:
-        config_js_path = os.path.join(
-            build_dir, 'juju-ui', 'assets', 'config.js')
-    render_to_file('config.js.template', context, config_js_path)
+    log('Generating the Juju GUI configuration file to {}.'.format(INI_PATH))
+    ini = read_config()
+    section = 'app:main'
+    ini.set(section, 'jujugui.sandbox', sandbox)
+    write_config(ini)
```

Comment:
It is but I suspect it'll need some fleshing out as we go forward. But not much I hope.